packages/service-library/src/servicelib/osparc_generic_resource.py

```diff
@@ -0,0 +1,112 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from servicelib.logging_utils import log_catch, log_context
+
+_logger = logging.getLogger(__name__)
+
+Ident = TypeVar("Ident")
+Res = TypeVar("Res")
+
+
+class BaseOsparcGenericResourceManager(ABC, Generic[Ident, Res]):
+    """Common interface used to manage the lifecycle of osparc resources.
+
+    An osparc resource can be anything that needs to be created and then removed
+    during the runtime of the platform.
+
+    Safe remove the resource.
+
+    For usage check ``packages/service-library/tests/test_osparc_generic_resource.py``
+    """
+
+    # If True ``get_or_create`` will add the `identifier` calling ``create``.
+    # Overwrite in constructor.
+    GET_OR_CREATE_INJECTS_IDENTIFIER: bool = False
```

Comment:
THOUGHT: To configure class policies I would separate it in a Config class like pydantic does. Nonetheless, having a flag to chance behaviour for such a simple wrapper somehow does not feel right